pymc_extras/statespace/core/properties.py

```diff
@@ -0,0 +1,257 @@
+from __future__ import annotations
+
+import warnings
+
+from collections.abc import Iterator
+from copy import deepcopy
+from dataclasses import dataclass, fields
+from typing import TYPE_CHECKING, Generic, Self, TypeVar
+
+from pymc_extras.statespace.utils.constants import (
+    ALL_STATE_AUX_DIM,
+    ALL_STATE_DIM,
+    OBS_STATE_AUX_DIM,
+    OBS_STATE_DIM,
+    SHOCK_AUX_DIM,
+    SHOCK_DIM,
+)
+
+if TYPE_CHECKING:
+    from pytensor.tensor.variable import TensorVariable
```

Comment:
You should be fine to import `TensorVariable` outside of the TYPE_CHECKING block. That is just for avoiding circular imports.